### PR TITLE
refactor shading proposal

### PR DESCRIPTION
**Pull request by igorbernstein2**
Reshuffling shading:
- use hbase's hbase-shaded-client by default: this will relocate hbase's older dependencies allowing us to avoid shading in standalone webapps and (in the future) beam
- create bigtable-hbase-1.x artifact that will always implement the latest hbase 1 interfaces. This should be the default artifact going forward. Other than hbase-shaded-client it uses no shading.
- create bigtable-hbase-1.x-shaded: this artifact will relocate bigtable dependencies as well. This can be used in environments where both hbase & bigtable dependencies cause conflicts. Mainly used for dataflow 1.x (which uses a version of protobuf that is both incompatible with bigtable and hbase)
- create bigtable-hbase-1.x-hadoop: the inverse of bigtable-hbase-1.x. It hides the bigtable dependencies and exposes hbase dependencies. This is meant to be used in hadoop.
- deprecate bigtable-hbase-1.0, bigtable-hbase-1.1, etc and encourage people to use  bigtable-hbase-1.x.

TODO:
- [x] fix dataflow 1.x usage of repackaged bigtable references ( com.google.bigtable.repackaged.com.google.com.google.bigtable vs com.google.bigtable.repackaged.com.google.bigtable and com.google.bigtable.repackaged.com.google.cloud vs com.google.bigtable.repackaged.com.google.cloud.bigtable)
- [ ] clean up // TODOs
- [x] possibly use bigtable-hbase-1.x in integration tests instead of the -hadoop variant

### CODE REVIEW

**Comment by sduskis** [comment]
Why does bigtable-hbase-1.0 depend on bigtable-hbase-shaded?

**Comment by sduskis** [comment]
Can you please pull out the explicit version numbers into a separate PR?

**Comment by igorbernstein2** [comment]
Working on splitting it up now.

**Comment by igorbernstein2** [comment]
rebase, this is ready for review

File: bigtable-hbase-parent/bigtable-hbase-shaded/pom.xml [hunk 1]
```diff
@@ -27,14 +27,104 @@ limitations under the License.
     <packaging>jar</packaging>
     <name>${project.groupId}:${project.artifactId}</name>
     <description>
+       DEPRECATED: Please use bigtable-hbase-1.x or bigtable-hbase-1.x-hadoop.
```
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase-shaded/pom.xml]: There should not be any users of bigtable-hbase-shaded.  Can you please delete the project if it's no longer referenced?

**Comment by igorbernstein2** [comment]
done

**Pull request merged**